backend/src/plugins/kube.ts

```diff
@@ -93,3 +102,16 @@ const getCurrentNamespace = async () => {
     }
   });
 };
+
+const getSAToken = async () => {
+  return new Promise<string>((resolve, reject) => {
+    if (currentContext === 'inClusterContext') {
```

Comment:
nope -- think of this flow. This didn't work before because what happens when you're not `currentContext` of `inClusterContext`?